### DISCUSSION TITLE
Non convex EBSD data set

### DISCUSSION THREAD

**Discussion by huseyinhzl** [mtex-toolbox/mtex#1684] [Ask Anything]
Hello,

I am new to this forum and MTEX. Recently, I have been trying to do parent grain reconstruction, and I have non-convex data set. I have read the documentation and applied the code recommended. I apply "grains = calcGrains(ebsdP('indexed'),'boundary','tight');" operation, I do not see any imaginary grains. However, when I perform job.votes.prob operation, I kept seeing unrelated grains at the top of my EBSD map, which I have not been able to solve. I have added the probability map to show the imaginary grains created. So, I am looking for help to solve this problem. 

![image](https://user-images.githubusercontent.com/120036622/206215858-688449f0-6102-4b33-bfaa-bfc299f01f18.png)

Thank you so much in advance

**Comment by kilir** [comment]
Hi,
`'tight'` comes with a numeric value with a default of 0.95, maybe try a lower one?
Cheers,
Rüdiger

**Reply by huseyinhzl** [reply]
Hi,

Thank you for the advice. When I changed the value to lower, it worked. However, I have realised that calculated grains are different than the previous one. With the previous settings, I was able to see grain boundary alpha, but now it is all white. It does not affect parent grain reconstruction though. 

![image](https://user-images.githubusercontent.com/120036622/206481333-62d3cf49-a2d3-43c4-9688-9bf664794794.png)

Can I ask one more question? I have been calculating the grains two times, during non convex hull operation and parent grain reconstruction with two different arguments (boundary and threshold). When I do so, grains are calculated as per older probability graph, but when I just calculate the grains during parent grain reconstruction with 'tight' operation, I get better results, so my question is should I calculate the grains two times or just once in parent grain reconstruction? If my question is not-understandable, I can share my coding
